### PR TITLE
PTRef: add meta-vj

### PR DESCRIPTION
**Pull request by pbougue**
links:
- meta-vj <--> vj
- meta-vj  ---> impact

### CODE REVIEW

File: source/ptreferential/tests/ptref_test.cpp [hunk 1]
```diff
@@ -503,14 +503,82 @@ static std::vector<nt::idx_t> query(nt::Type_e requested_type, std::string reque
...
+    builder.vj("B", "1011")("stop3", "10:00"_t);
+    builder.vj("C", "1000")("stop3", "10:00"_t);
+    builder.finish();
+    nt::idx_t a = 0;
+    nt::idx_t b = 1;
+    nt::idx_t c = 2;
+
+    auto check = [](const std::vector<nt::idx_t>& col, std::set<nt::idx_t> s_ref) {
+        std::set<nt::idx_t> s_col(std::begin(col), std::end(col));
+        BOOST_CHECK_EQUAL(s_col, s_ref);
```
Comment (antoine-de) [inline source/ptreferential/tests/ptref_test.cpp]: why not use BOOST_RANGE_EQUALS  (or whatever the name of the macro is) ?

File: source/ptreferential/tests/ptref_test.cpp [hunk 2]
```diff
@@ -503,14 +503,82 @@ static std::vector<nt::idx_t> query(nt::Type_e requested_type, std::string reque
...
+
+    // looking for MetaVJ 0 another way
+    Filter filter;
+    filter.navitia_type = Type_e::MetaVehicleJourney;
+    filter.attribute = "uri";
+    filter.op = EQ;
+    filter.value = "vehicle_journey 0";
+    indexes = get_indexes<navitia::type::MetaVehicleJourney>(filter, Type_e::MetaVehicleJourney, *(builder.data));
+    BOOST_REQUIRE_EQUAL(indexes.size(), 1);
+    BOOST_CHECK_EQUAL(indexes[0], 0);
```
Comment (xlqian) [inline source/ptreferential/tests/ptref_test.cpp]: Maybe `BOOST_RANGE_EQUALS` to replace `BOOST_REQUIRE_EQUAL` and `BOOST_CHECK_EQUAL` ?

**Comment by antoine-de** [comment]
OK for me

**Comment by xlqian** [comment]
Ok for me

**Pull request merged**